### PR TITLE
Add masked input component

### PR DESCRIPTION
**Pull request by elmarques**
# What

This PR adds the masked input component to the project.

# Why

Users can be guided by masks to avoid entering incorrect data in the form fields.

# How

* Adds 2 libs to the components subproject: for masks and number formatting
* Creates a wrapper component `InputMasked` to grouping the masking logic
* Adds the global method `mountWithTheme` in the Jest setup file _(existing methods didn't work well with the added libs due to DOM event handling)_
* Add usage examples in the documentation

# Sample

Captura de Tela 2020-03-23 às 10 19 20

# Refs

* [Clubhouse card](https://app.clubhouse.io/magnetis/story/23899/criar-input-com-máscara-no-galaxy) [ch23899]
* [Masking lib](https://www.npmjs.com/package/react-input-mask)
* [Number formatting lib](https://www.npmjs.com/package/react-number-format)

### CODE REVIEW

File: jest.setup.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ global.rendererCreateWithTheme = Tree => {
   return renderer.create(<AstroThemeProvider theme={theme} children={Tree} />);
 };
 
+global.mountWithTheme = Tree => {
+  return mount(<AstroThemeProvider theme={theme} children={Tree} />);
+};
+
```
Comment (AgtLucas) [inline jest.setup.js]: Nice!

File: packages/components/src/InputText/index.story.js [hunk 2]
```diff
@@ -1,12 +1,11 @@
-import React from 'react';
 import { storiesOf } from '@storybook/react';
+import React from 'react';
 import InputText from './index.js';
 
 storiesOf('InputText', module)
-  .add('default', () => <InputText m={8} inputId="input0" labelId="label0" />)
```
Comment (AgtLucas) [inline packages/components/src/InputText/index.story.js]: Is it ok to remove this margin?

File: packages/components/src/InputMasked/index.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  };
+
+  return maskType === maskTypes.currency ? (
+    <NumberFormat
+      {...commonProps}
+      customInput={InputText}
+      placeholder={maskType.placeholder}
+      thousandSeparator="."
+      decimalSeparator=","
+      decimalScale={2}
```
Comment (AgtLucas) [inline packages/components/src/InputMasked/index.js]: Is it possible by any chance for us to have a case where the `decimalScale` should be other value other than `2`? 

Maybe we should allow custom `decimalScale` via props, what do you think?

File: packages/components/src/InputText/index.story.js (hunk 2 above)
Comment (elmarques) [reply packages/components/src/InputText/index.story.js]: Yes. We created this property to move the input away from the edges of the Storybook sandbox.

It didn't happen. Also, I realized that all the other components don't do this.

So I proposed to remove this dead code in name of the ~OCD~ standards. 😝

File: packages/components/src/InputMasked/index.js (hunk 3 above)
Comment (elmarques) [reply packages/components/src/InputMasked/index.js]: Nice! I haven't seen it until today, but it's always possible. I added the parameter: 3509231

**Pull request merged**